src/lowering/desugar.ml

```diff
@@ -438,8 +438,9 @@ and export_footprint self_id expr =
   ([ letD (var v typ) (
        funcE v (Shared Query) Promises [bind1] [] [ret_typ] (
            (asyncE T.Fut bind2
-              (blockE [expD (assertE (primE (I.RelPrim (caller, Operator.EqOp))
-                                        [primE I.ICCallerPrim []; selfRefE caller]));
+              (blockE [expD (assertE (orE (primE (I.RelPrim (caller, Operator.EqOp))
```

Comment:
Might be cheaper to call ICCallerPrim just once, but don't think it matters.

Reply:
I am not overly concerned :-)

Reply:
This is now #4108!